### PR TITLE
Add batched_copy to All Communicators

### PR DESCRIPTION
**Pull request by belldandyxtq**
This commit does following adjustments:
1. moving the batched_copy from pure_nccl_communicator to the
base class of communicators, which enables batched_copy in all the
communicators.
2. A batched_copy parameter is added to the __init__ of all the
communicators.
3. The warning when using batched_copy on creating communicators
other than pure_nccl_communicator is removed.

### CODE REVIEW

**Comment by belldandyxtq** [comment]
pfnci, test this please

**Comment by shu65** [comment]
Could you resolve conflicts?

File: chainermn/communicators/pure_nccl_communicator.py [hunk 1]
```diff
@@ -196,160 +166,16 @@ def _multi_node_mean_nccl(self, sendbuf, recvbuf,
...
-            float* src = (float*) src0;
-            if (dst_dtype == NCCL_FLOAT16) {
-                ((half*) (params_dptr[j]))[dst_idx] = (half) src[tid];
-            }
-            else if (dst_dtype == NCCL_FLOAT32) {
-                ((float*) (params_dptr[j]))[dst_idx] = (float) src[tid];
-            }
-       }
-    }''', 'cupy_batched_unpack_params')
+def _get_converting_kernel(src_dtype, dst_dtype, kernel_name):
```
Comment (shu65) [inline chainermn/communicators/pure_nccl_communicator.py]: Could you remove `_get_converting_kernel()`, `_get_param_data_dtype()`, and `_get_param_grad_dtype()` ?
These functions are not used now.
These functions were deleted in other PR

**Comment by shu65** [comment]
pfnci, test this please

**Pull request merged**